omero/annotation_scripts/Import_from_csv.py

```diff
@@ -0,0 +1,858 @@
+            for target_obj in target_obj_l:
+                name = get_obj_name(target_obj)
+                assert name not in target_d.keys(), \
+                    ("Target objects identified by name have at " +
+                     f"least one duplicate: {name}")
+                target_d[name] = target_obj
+        else:
+            # Setting the dictionnary target_id:target_obj
+            # keys as string to match CSV reader output
+            target_d = {str(target_obj.getId()): target_obj
+                        for target_obj in target_obj_l}
+        ntarget_processed += len(target_d)
+
+        if tag_d is None and "tag" in [h.lower() for h in header]:
+            # Create the tag dictionary a single time if needed
+            tag_d, tagset_d, tagtree_d, tagid_d = get_tag_dict(
+                conn, use_personal_tags
+            )
+        # Replace the tags in the CSV by the tag_id to use
+        rows, tag_d, tagset_d, tagtree_d, tagid_d = preprocess_tag_rows(
+            conn, header, rows, tag_d, tagset_d, tagtree_d, tagid_d,
+            create_new_tags, split_on
+        )
+
+        ok_idxs = [i for i in range(len(header)) if i not in cols_to_ignore]
+        for row in rows:
+            # Iterate the CSV rows and search for the matching target
+            target_id = row[idx_id]
+            # skip empty rows
+            if target_id == "":
+                continue
+            if target_id in target_d.keys():
+                target_obj = target_d[target_id]
+                # add name/id to processed set
+                if file_ann_multiplied:
+                    processed_names.add(target_id)
+            else:
+                # add name/id to missing set
+                if file_ann_multiplied:
+                    missing_names.add(target_id)
+                else:
+                    total_missing_names += 1
+                    print(f"Not found: {target_id}")
+                continue
+
+            if split_on != "":
+                parsed_row, parsed_ns, parsed_head = [], [], []
+                for i in ok_idxs:
+                    curr_vals = row[i].strip().split(split_on)
+                    parsed_row.extend(curr_vals)
+                    parsed_ns.extend([namespaces[i]] * len(curr_vals))
+                    parsed_head.extend([header[i]] * len(curr_vals))
+            else:
+                parsed_row = [row[i] for i in ok_idxs]
+                parsed_ns = [namespaces[i] for i in ok_idxs]
+                parsed_head = [header[i] for i in ok_idxs]
+
+            updated = annotate_object(
+                conn, target_obj, parsed_row, parsed_head,
+                parsed_ns, exclude_empty_value, tagid_d, split_on
+            )
+
+            if updated:
+                if result_obj is None:
+                    result_obj = target_obj
+                ntarget_updated += 1
+                ntarget_updated_curr += 1
+
+        if ntarget_updated_curr > 0 and attach_file:
```

Comment:
I'm not sure of the logic here, but I chose to attach the file, then ran the script but forgot to choose the correct target so it "failed" didn't get attached.
Each time this happens we end up with an orphaned FileAnnotations since it doesn't get deleted. And the user can't delete Orphaned FileAnnotations without attaching them first (at least in the webclient).
I would almost consider *always* attaching the FileAnnotations, unless there is a particular reason to give the user a choice?

Reply:
That was a misunderstanding from me, I thought those were considered temporary files. The logic was to not attach the file if it doesn't work, to avoid the extra step of deleting the "bad csv". I don't see an issue for removing that choice. Now with all other parameters, it will clear the script a little. 

But the FileAnnotation could be deleted too, right? 
Thinking about not having the permissions to annotate an object (we should clean up the uploaded csv)